src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -114,6 +115,7 @@
     @CompilationFinal private SecureRandom random;
     private final Hashing hashing;
     private final RopeCache ropeCache;
+    private final PathToRopeCache pathToRopeCache = new PathToRopeCache(this);
```

Comment:
I was thinking this would look more like:

`private final RopeCache pathToRopeCache = new RopeCache(...)`

But this works, too.

Reply:
They have different interface, the RopeCache doesn't need j.l.String as input so it's simpler/clearer to split.